news-sentiment/README.md

```diff
@@ -0,0 +1,78 @@
+ 
+## Technology highlights
+
+- The sheet provides an extra menu option: **News Headlines Sentiments** and when clicked renders the option: **Analyze News headlines...**.  When selected, it calls the `showNewsPrompt()` function and triggers a popup dialog to allow the user to enter their own news topic of choice. 
+- After entering a topic, the function `analyzeNewsHeadlines()` executes and calls the following key functions:
+   * `reformatSheet()` Clears and reformats the sheet.
+   * `getHeadlinesArray()` Fetches a fresh set of headlines by passing the user entered topic to the News API. It then syncs the returned headlines to the sheet.
+   * `getSentiments()` Iterates through each headline and calls the Natural Language API to obtain a sentiment value and then updates the sheet with values.
+
+## Try it
+
+### Pre-Steps - Obtaining your API keys
+To run this solution you will need to obtain 2 API keys, one from the [Google Cloud Natural Language API](https://cloud.google.com/natural-language), and the second from the free News API @ http://newsapi.org/.
+
+1.  **Google** - Configure a GCP Project to obtain an API key for the Natural Language API.
+   * Create a new or use an existing GCP project: https://console.cloud.google.com 
+   * If creating a new project: 
+      * Select your associated Billing account.
+      * Accept the defaults for Organization and Location.
+      * Click CREATE, and then select the new project in the console.
+   * Enable the Google Cloud Natural Language API on your GCP project by going to **APIs and Services** dashboard.
+      * Click + ENABLE APIS AND SERVICES
+      * Search for 'Cloud Natural Language API', and enable it.
+   * Click 'Credentials' on the left side menu and click + CREATE CREDENTIALS
+      * Select 'Api key' in the drop down menu. 
+      * Save this key to add to your Apps Script `Code.gs` file. 
+
+1.  **News API** - Register for a free News API key.
+   * Register for a free News API account at: https://newsapi.org/
+   * Click 'Get API Key' and follow the steps.
+   * Save the key to add to your Apps Script `Code.gs` file. 
+
+### Pre-Steps - Customizing your copy of the solution
+Follow these steps to copy and customize the solution with your API keys.
+
+1. Make a copy of example Google Sheet and code.
+   [by clicking this link](https://docs.google.com/spreadsheets/d/1XwxcB9W9dkNBu8sSag3cwvQxGLNAeMCxXcfjm9zS9H4/copy)
+   in your browser. It automatically includes Google the Apps Script code
+   you need.
+
+1. Open the associated Apps Script Project from the Sheets menu: `Extensions -> Apps Script`.
+1. Update the `Code.gs` script with your API keys:
+   - const googleAPIKey = `YOUR_GOOGLE_API_KEY`;
+   - const newsApiKey = `YOUR_NEWS_API_KEY`;
+1. Save the script and return to the Spreadsheet.
+
+### Running the News Sentiment Analyzer
+1. Click the `News Headlines Sentiments` custom menu option at the top of the sheet.
+1. Select 'Analyze News Headlines...'
```

Comment:
Lots of UI elements in this section, make sure they're all bolded instead of quoted.

Reply:
Done.